### PR TITLE
Prettify the frontend code

### PR DESCRIPTION
**Pull request by etr2460**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [x] Build / Development Environment
- [ ] Documentation

### SUMMARY
Adds prettier to all of our frontend code. Retains the settings from dashboard code where Prettier was already applied. Otherwise use the default settings.

The first commit is my manual changes, the second commit is running `npm run lint-fix` and cleaning up some unnecessary concated strings.

Enforced via a lint rule, either setup an autoformatter in your IDE or run `npm run lint-fix` prior to committing to format.

I'm planning to rebase this and merge is sometime on Wednesday afternoon (if no one disagrees) so that it doesn't affect too many other PRs (assuming people aren't working over Thanksgiving)

### TEST PLAN
Manual testing in Dashboard, Explore, and SQL Lab. Also CI

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@williaster @kristw @graceguo-supercat @rusackas @nytai @betodealmeida

### CODE REVIEW

**Comment by rusackas** [comment]
I'm happy to see the scattered `.prettierrc` files disappear... I almost did that, but wasn't sure the history/emotions attached to those.

In the remaining `.prettierrc` I see the rules are to go with single quotes, and trailing commas. I like the trailing commas for commit/blame history. For quotes, I'd be inclined to go with Prettier's defaults, unless people feel passionate about those single quotes.

**Comment by etr2460** [comment]
Yeah, I definitely would prefer to keep with the prettier defaults. I kept single quotes because that's what we have everywhere already, but i'm happy to drop that too

**Comment by etr2460** [comment]
removed single quotes, but will readdress if anyone is passionate

**Review by rusackas — APPROVED** [review]
I think we might need more voices on the double/single quotes issue, since it's so far-reaching.

*Pros:*
• Prettier gets to act like Prettier wants to.
• Double quotes are kind of old school, and I like them, personally.
• You'll rocket upward on the repo's leaderboards ;)

*Cons:* 
• Soooo many lines affected means this'll basically wallop the Blame history for the repo. People may feel more strongly about that then some pedantic argument about single/double quotes. @mistercrunch might want to weigh in here.

**Comment by graceguo-supercat** [comment]
i don't like or dislike double quotes. My question is why do you want to change it from single quotes to double quotes?

File: superset/assets/.prettierrc [hunk 1]
```diff
@@ -1,4 +1,3 @@
 {
-  "singleQuote": true,
```
Comment (graceguo-supercat) [inline superset/assets/.prettierrc]: why remove this singleQuote rule?

**Comment by graceguo-supercat** [comment]
Sorry i just realized you prefer prettier default setting over adding a rule. 
But I would rather extend existed rule than change hundreds of lines.

**Comment by etr2460** [comment]
Yup, there's pros and cons to both sides. Looks like @rusackas and I prefer Prettier defaults, and @graceguo-supercat prefers keeping the same. Let's try to get input from @mistercrunch, @betodealmeida, and/or @khtruong to try and find consensus?

Then I'll either revert the last commit or keep it depending on the result

**Comment by khtruong** [comment]
I am fine with either. I am just a fan of consistency.

**Comment by kristw** [comment]
I personally prefer single quote. All `@superset-ui` repos (`superset-ui`, `superset-ui-plugins`, `superset-ui-plugins-deckgl`, `superset-ui-template`, ...) currently use single quote, which would be great to keep same consistency. It would also help Airbnb developers as Airbnb style guide prefers single quote and our internal repos all use that.

All `@superset-ui` repos share same build configuration (via `beemo`) with the following `prettier` overrides. 
https://github.com/apache-superset/build-config/blob/master/configs/prettier.js

**Comment by williaster** [comment]
I plus one the single quote style

**Comment by rusackas** [comment]
The more I think on it, the more I'm coming around to single quotes... if for no other reason, just to keep git blame/history more intact.

**Comment by etr2460** [comment]
Single quote it is! I'll revert the double quote PR and merge this afternoon

**Comment by williaster** [comment]
This will be great overall for frontend code, thanks for adding Erik!

**Pull request merged**